contracts/utils/EscrowLib.sol

```diff
@@ -0,0 +1,258 @@
+pragma solidity 0.8.9;
+
+import {Denominations} from "@chainlink/contracts/src/v0.8/Denominations.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IOracle} from "../interfaces/IOracle.sol";
+import {ILineOfCredit} from "../interfaces/ILineOfCredit.sol";
+import {IEscrow} from "../interfaces/IEscrow.sol";
+import {CreditLib} from "../utils/CreditLib.sol";
+import {LineLib} from "../utils/LineLib.sol";
+
+struct EscrowState {
+    address line;
+    address[] collateralTokens;
+    /// if lenders allow token as collateral. ensures uniqueness in collateralTokens
+    mapping(address => bool) enabled;
+    /// tokens used as collateral (must be able to value with oracle)
+    mapping(address => IEscrow.Deposit) deposited;
+}
+
+library EscrowLib {
+    using SafeERC20 for IERC20;
+
+    // return if have collateral but no debt
+    uint256 constant MAX_INT =
+        115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    function isLiquidatable(EscrowState storage self, address oracle, uint256 minimumCollateralRatio) external returns(bool) {
+      return _getLatestCollateralRatio(self, oracle) < minimumCollateralRatio;
+    }
+
+    function updateLine(EscrowState storage self, address _line) external returns(bool) {
+      require(msg.sender == self.line);
+      self.line = _line;
+      return true;
+    }
+
+    /**
+     * @notice updates the cratio according to the collateral value vs line value
+     * @dev calls accrue interest on the line contract to update the latest interest payable
+     * @return the updated collateral ratio in 18 decimals
+     */
+    function _getLatestCollateralRatio(EscrowState storage self, address oracle) public returns (uint256) {
+        (uint256 principal, uint256 interest) = ILineOfCredit(self.line).updateOutstandingDebt();
+        uint256 debtValue =  principal + interest;
+        uint256 collateralValue = _getCollateralValue(self, oracle);
+        if (collateralValue == 0) return 0;
+        if (debtValue == 0) return MAX_INT;
+
+        return _percent(collateralValue, debtValue, 18);
```

Comment:
this is the only place we use _percent. when this was an internal function in the contract i dont think there was as big a gas difference but now its public so it would be more efficient to include the _percent code directly in _getLatestCollateralRatio. 

Can you do the same for _addSpigot and _operate in SpigotLib too please.